applications/StructuralMechanicsApplication/python_scripts/class_StructuralMechanics.py

```diff
@@ -0,0 +1,268 @@
+        else:
+            self.time = start_time
+            self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] = 0
+
+        if (self.parallel_type == "OpenMP") or (KratosMPI.mpi.rank == 0):
+            KratosMultiphysics.Logger.PrintInfo("::[KSM Simulation]:: ", "Analysis -START- ")
+
+    def ExecuteInitializeSolutionStep(self):
+        """ Initialize the timestep and advance in time. Called once per timestep """
+        self.time += self.delta_time
+        self.main_model_part.ProcessInfo[KratosMultiphysics.STEP] += 1
+        self.main_model_part.CloneTimeStep(self.time)
+
+        if (self.parallel_type == "OpenMP") or (KratosMPI.mpi.rank == 0):
+            KratosMultiphysics.Logger.PrintInfo("STEP: ", self.main_model_part.ProcessInfo[KratosMultiphysics.STEP])
+            KratosMultiphysics.Logger.PrintInfo("TIME: ", self.time)
+
+        for process in self.list_of_processes:
+            process.ExecuteInitializeSolutionStep()
+
+        if (self.output_post == True):
+            self.gid_output.ExecuteInitializeSolutionStep()
+
+    def ExecuteBeforeSolve(self):
+        """ Function to be called before solving. Can be executed several times per timestep """
+        pass
+
+    def SolveSolutionStep(self):
+        """ Solving one step. Can be called several times per timestep """
+        self.ExecuteBeforeSolve()
+        self.solver.Solve()
+        self.ExecuteAfterSolve()
+
+    def ExecuteAfterSolve(self):
+        """ Function to be called after solving. Can be executed several times per timestep """
+        pass
+
+    def ExecuteFinalizeSolutionStep(self):
+        """ Finalizing the timestep and printing the output. Called once per timestep """
+        for process in self.list_of_processes:
+            process.ExecuteFinalizeSolutionStep()
+
+        if (self.output_post == True):
+            self.gid_output.ExecuteFinalizeSolutionStep()
+
+        for process in self.list_of_processes:
+            process.ExecuteBeforeOutputStep()
+
+        if (self.output_post == True) and (self.gid_output.IsOutputStep()):
+            self.gid_output.PrintOutput()
+
+        for process in self.list_of_processes:
+            process.ExecuteAfterOutputStep()
+
+        self.solver.SaveRestart() # whether a restart-file is written is decided internally
+
+    def ExecuteFinalize(self):
+        """ Operations to be performed at the end of the Analysis """
+        for process in self.list_of_processes:
+            process.ExecuteFinalize()
+
+        if (self.output_post == True):
+            self.gid_output.ExecuteFinalize()
+
+        if (self.parallel_type == "OpenMP") or (KratosMPI.mpi.rank == 0):
+            KratosMultiphysics.Logger.PrintInfo("::[KSM Simulation]:: ", "Analysis -END- ")
+
+
+if __name__ == "__main__":
+    from sys import argv
+
+    if len(argv) > 2:
+        raise Exception("Wrong number of input arguments")
+
+    if len(argv) == 2: # ProjectParameters is being passed from outside
```

Comment:
May be also printing how to run and what are expected as arguments here can be helpful and informative ??